packages/care-package/abstract-plugin/src/ProviderOrg.ts

```diff
@@ -36,110 +52,90 @@ export class ProviderOrg {
-        };
+        return vdcStorageProfiles.name;
     }
 
-    public async getProviderOrgEntities(): Promise<ProviderOrgEntities> {
-        const legacyApiClient = this.apiConfig.makeLegacyApiClient();
-        const organisation = await ProviderOrg.getOrg(legacyApiClient);
-        const metadata = await ProviderOrg.getMetadata(legacyApiClient, organisation.id);
-
-        const vdcMetadata = metadata.metadataEntry
-            .find(entry => entry.key === 'ProviderOrgVdc')
-            .typedValue;
-        const networkMetadata = metadata.metadataEntry
-            .find(entry => entry.key === 'ProviderOrgNetwork')
-            .typedValue;
-        const storageProfileMetadata = metadata.metadataEntry
-            .find(entry => entry.key === 'ProviderOrgVdcStorageProfile')
-            .typedValue;
-
-        const providerOrgApiConfig = this.apiConfig.actAs(organisation.id);
-        return {
-            org: organisation,
-            vdc: await ProviderOrg.getVdc(providerOrgApiConfig, vdcMetadata),
-            network: await ProviderOrg.getNetwork(providerOrgApiConfig, networkMetadata),
-            storageProfile: await ProviderOrg.getStorageProfile(providerOrgApiConfig, storageProfileMetadata),
-            limits: ''
-        };
+    public async getProviderOrgInterface(): Promise<ProviderOrgInterface> {
+        log('Getting ProviderOrg Interface');
+        const definedEntityTypeApiClient = this.apiConfig.makeApiClient(DefinedEntityTypeApi);
+        const definedEntityTypes = await definedEntityTypeApiClient
+            .getDefinedEntityTypes(1, 1, 'nss==solutions_organization_provider', '', 'version');
+        const definedEntityType = definedEntityTypes.body.values.pop();
+
+        if (!definedEntityType) {
+            throw new Error('Provider Organization Interface is not configured.');
+        }
+        log('DefinedEntityType of ProviderOrg Interface found.');
+
+        const definedEntityTypeClient = this.apiConfig.makeApiClient(DefinedEntityApi);
+        const definedEntities = await definedEntityTypeClient.getDefinedEntitiesByEntityType(
+            definedEntityType.vendor,
+            definedEntityType.nss,
+            definedEntityType.version,
+            1,
+            1);
+
+        if (!definedEntities) {
+            throw new Error('Provider Organization Interface is not configured.');
+        }
+        log('DefinedEntity of ProviderOrg Interface found.');
+
+        const definedEntity = definedEntities.body.values.pop();
+        const providerOrgInterface = definedEntity.entity as ProviderOrgInterface;
+        const providerOrgApiConfig = this.apiConfig.actAs(providerOrgInterface.organization.id);
+        providerOrgInterface.organization.name = await ProviderOrg.getOrg(this.apiConfig, providerOrgInterface.organization.id);
+        log(`Organization: ${providerOrgInterface.organization.name} found.`);
+        for (const vdc of providerOrgInterface.organization.vdcs) {
```

Comment:
Can't we either fetch these in parallel or use query by all IDs?
Same for all for loops bellow ?